### PR TITLE
[Feature] Redesign Filter

### PR DESCRIPTION
**Pull request by KobbyMmo**
## Description

This PR adds Filter function

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Desktop Screenshots
image

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Quick feedback:

1. Click an item is clicked multiple times, it shouldn't add itself more than once
Screenshot from 2022-01-18 18-02-09
2. When filtering by STATUS, either items should allow only one item to be selected at any given time or do an OR. Right now we're doing AND and it's impossible for any promise to have 2 statuses at the same time.

File: src/components/Promises/index.js [hunk 1]
```diff
@@ -26,44 +26,67 @@ function Promises({
...
+      filters.push(...filterStatusItems);
+    } else {
+      filters.push(...filterCategoryItems);
+    }
+    return filters.find((f) => f.slug === slug);
+  };
+  const handleFilterClick = (slug) => {
+    setSelectedFilters((prev) => {
+      prev.find((f) => f.slug === slug);
+      if (prev.find((f) => f.slug === slug)) {
```
Comment (kilemensi) [inline src/components/Promises/index.js:45]: Why are we repeating this?

File: src/components/Promises/index.js [hunk 2]
```diff
@@ -26,44 +26,67 @@ function Promises({
...
 
   useEffect(() => {
     if (withFilter) {
-      const selectedStatuses = statusesFilters
-        ?.filter((filter) => filter.active)
-        .map((filter) => filter.slug);
       const hasStatus = (item) => {
         const promiseSlug = slugify(item.status.title);
-        return selectedStatuses.every((c) => c === promiseSlug);
+        return selectedFilters?.some((c) => c.slug === promiseSlug);
```
Comment (kilemensi) [inline src/components/Promises/index.js:71]: No need for `?.` We're initializing `selectedFilters` with `[]`

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

Does sort work?

Screenshot from 2022-01-19 11-44-43

**Comment by KobbyMmo** [comment]
> Does sort work?

Works but we have to rethink the meaning of a promise deadline. Currently, the promise doesn't have a deadline property, this was because check didn't support it, and we were using events that could be empty. We therefore should allow a promise deadline property and use the events. @kilemensi

**Comment by kilemensi** [comment]
> > Does sort work?
> 
> Works but we have to rethink the meaning of a promise deadline. Currently, the promise doesn't have a deadline property, this was because check didn't support it, and we were using events that could be empty. We therefore should allow a promise deadline property and use the events. @kilemensi

Not sure I get @KobbyMmo ... gSheets also supports events so theoretically we can still use events for deadline.

What am I missing?

**Comment by KobbyMmo** [comment]
> What am I missing?

What i am saying is should we track the promised deadline with the events or a separate deadline field. I feel we were forced to use events because of checks. but if we still want to maintain the same structure, then fine. So that means the event should also be required. @kilemensi

**Comment by kilemensi** [comment]
> > What am I missing?
> 
> What i am saying is should we track the promised deadline with the events or a separate deadline field. I feel we were forced to use events because of checks. but if we still want to maintain the same structure, then fine. So that means the event should also be required. @kilemensi

Cool @KobbyMmo ... add a story in PT for us to discuss this further outside this PR.

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**